test/hotspot/jtreg/compiler/c2/cr7200264/TestIntVect.java

```diff
@@ -21,622 +21,763 @@
+        int[] a4 = new int[ARRLEN];
+        long[] p2 = new long[ARRLEN/2];
+        int gold_sum = 0;
+        for (int i=0; i<ARRLEN; i++) {
+            int val = (int)(ADD_INIT+i);
+            gold_sum += val;
+            a1[i] = val;
+            a2[i] = (int)VALUE;
+            a3[i] = (int)-VALUE;
+            a4[i] = (int)BIT_MASK;
+        }
+
+        System.out.println("Warmup");
+        for (int i=0; i<ITERS; i++) {
+            test_sum(a1);
+            test_addc(a0, a1);
+            test_addv(a0, a1, (int)VALUE);
+            test_adda(a0, a1, a2);
+            test_subc(a0, a1);
+            test_subv(a0, a1, (int)VALUE);
+            test_suba(a0, a1, a2);
+            test_mulc(a0, a1);
+            test_mulv(a0, a1, (int)VALUE);
+            test_mula(a0, a1, a2);
+            test_divc(a0, a1);
+            test_divv(a0, a1, (int)VALUE);
+            test_diva(a0, a1, a2);
+            test_mulc_n(a0, a1);
+            test_mulv(a0, a1, (int)-VALUE);
+            test_mula(a0, a1, a3);
+            test_divc_n(a0, a1);
+            test_divv(a0, a1, (int)-VALUE);
+            test_diva(a0, a1, a3);
+            test_andc(a0, a1);
+            test_andv(a0, a1, (int)BIT_MASK);
+            test_anda(a0, a1, a4);
+            test_orc(a0, a1);
+            test_orv(a0, a1, (int)BIT_MASK);
+            test_ora(a0, a1, a4);
+            test_xorc(a0, a1);
+            test_xorv(a0, a1, (int)BIT_MASK);
+            test_xora(a0, a1, a4);
+            test_sllc(a0, a1);
+            test_sllv(a0, a1, VALUE);
+            test_srlc(a0, a1);
+            test_srlv(a0, a1, VALUE);
+            test_srac(a0, a1);
+            test_srav(a0, a1, VALUE);
+            test_sllc_n(a0, a1);
+            test_sllv(a0, a1, -VALUE);
+            test_srlc_n(a0, a1);
+            test_srlv(a0, a1, -VALUE);
+            test_srac_n(a0, a1);
+            test_srav(a0, a1, -VALUE);
+            test_sllc_o(a0, a1);
+            test_sllv(a0, a1, SHIFT);
+            test_srlc_o(a0, a1);
+            test_srlv(a0, a1, SHIFT);
+            test_srac_o(a0, a1);
+            test_srav(a0, a1, SHIFT);
+            test_sllc_on(a0, a1);
+            test_sllv(a0, a1, -SHIFT);
+            test_srlc_on(a0, a1);
+            test_srlv(a0, a1, -SHIFT);
+            test_srac_on(a0, a1);
+            test_srav(a0, a1, -SHIFT);
+            test_pack2(p2, a1);
+            test_unpack2(a0, p2);
+            test_pack2_swap(p2, a1);
+            test_unpack2_swap(a0, p2);
+        }
+
+        // Test and verify results
+        System.out.println("Verification");
+        int errn = 0;
+        {
+            int sum = test_sum(a1);
+            if (sum != gold_sum) {
+                System.err.println("test_sum:  " + sum + " != " + gold_sum);
+                errn++;
+            }
+
+            test_addc(a0, a1);
+            for (int i=0; i<ARRLEN; i++) {
+                errn += verify("test_addc: ", i, a0[i], (int)((int)(ADD_INIT+i)+VALUE));
```

Comment:
When translating the test, I focused only on translating the old ad-hoc IR tests to the IR verification framework. Do we want to extend the scope of this issue to also update the testing code? Mainly, the reason I'm asking is that copies of `TestIntVect.java` appears in many places (e.g., `test/compiler/6340864/TestIntVect.java`), and the testing pattern in particular appears in many places (grep for, e.g., `errn += verify(`). If we update the pattern here, we should probably also update it everywhere else?

Reply:
Good point, I guess it's okay then to leave this code as as it is. It just looked odd to do correctness testing the way it does. But as you say, if you change that you probably also need to revisit other tests. That's probably not worth it. The main goal should be to introduce IR matching with the IR framework. So, I'm fine with not touching that code or the print statements.

Reply:
Would it perhaps be useful to create a separate RFE for your cleanup suggestions?

Reply:
What I think might eventually be much more helpful, is to add result verification into the IR framework, and not have to hand-code it everywhere. That would be worth investing some time in, because it would simplify future testing, and make sure that people generally have to do result verification (it gets forgotten too easily).

Reply:
> Would it perhaps be useful to create a separate RFE for your cleanup suggestions?

Not sure if it's worth it. But cleaning up tests is always a good thing.

> What I think might eventually be much more helpful, is to add result verification into the IR framework, and not have to hand-code it everywhere. That would be worth investing some time in, because it would simplify future testing, and make sure that people generally have to do result verification (it gets forgotten too easily).

That's a good point. We should provide some support for that at some point or even enforce it.